[RFT]x86: rework image builder code

aparcar opened the pull request:
The x86 target still uses legacy code to create the images instead of
making use of image.mk.

This patch is a rebase of Tomasz previous patch series, excluding the
bootloader upgrade bits[0].

* x86/grub2: move grub2 image creation to grub2 package

> Let the grub2 package take care of creating installable grub2 images,
> this will allow creating grub2 images without first calling x86 image
> generation recipe. Also as side effect, since those images are now
> shared, it'll reduce the number of calling grub-mkimage.

* x86: switch image generation to new code

> * Split up image generation recipe to smaller chunks to make it more
>   generic and reusable.
> * Make iso images x86 specific and drop their definition as root
>   filesystem.
> * Convert image creation process to generic code specified in image.mk.
> * Make geode subtarget inherit features from the main target instead of
>   redefining them.
> * For subtargets create device definitions with basic packages set.

* x86: remove obsolete profiles

> Rely on device profiles instead for packages selection.

CC: Tomasz Maciej Nowak <tomek_n@o2.pl>

[0]: https://patchwork.ozlabs.org/cover/1024165/

Signed-off-by: Paul Spooren <mail@aparcar.org>

ynezz commented on target/linux/x86/image/Makefile:
```diff
@@ -1,5 +1,6 @@
 # 
 # Copyright (C) 2006-2012 OpenWrt.org
+# Copyright (C) 2019 Alexander Couzens <lynxis@fe80.eu>
```
While at it, you could add all authors:

```
git log --format="%aN <%aE>" target/linux/x86/image/Makefile | sort -u
```

Really, we should get rid of this nonsense, not encourage it. We've VCS for authorship records.

aparcar replied:
If @lynxis don't mind I remove the line

ynezz commented on target/linux/x86/image/Makefile:
```diff
@@ -46,136 +47,127 @@ ROOTPART:=$(if $(ROOTPART),$(ROOTPART),PARTUUID=$(IMG_PART_SIGNATURE)-02)
-  endef
+# Convert a raw image into a $1 type image.
+# E.g. | qemu-image vdi
+define Build/qemu-image
```
this is perhaps generic material, so it should be moved into include/image* ?

tmn505 commented:
Not a review but a small hint on what I did in the past: https://patchwork.ozlabs.org/cover/1024165
especially "x86: switch image generation to new code". Maybe some ideas there could be helpful.

aparcar committed:
x86: switch to generic image generate code

The x86 target still uses legacy code to create the images instead of
making use of image.mk.

This patch is a rebase of Tomasz previous patch series, excluding the
bootloader upgrade bits[0].

* x86/grub2: move grub2 image creation to grub2 package

> Let the grub2 package take care of creating installable grub2 images,
> this will allow creating grub2 images without first calling x86 image
> generation recipe. Also as side effect, since those images are now
> shared, it'll reduce the number of calling grub-mkimage.

* x86: switch image generation to new code

> * Split up image generation recipe to smaller chunks to make it more
>   generic and reusable.
> * Make iso images x86 specific and drop their definition as root
>   filesystem.
> * Convert image creation process to generic code specified in image.mk.
> * Make geode subtarget inherit features from the main target instead of
>   redefining them.
> * For subtargets create device definitions with basic packages set.

* x86: remove obsolete profiles

> Rely on device profiles instead for packages selection.

CC: Tomasz Maciej Nowak <tomek_n@o2.pl>

[0]: https://patchwork.ozlabs.org/cover/1024165/

Signed-off-by: Paul Spooren <mail@aparcar.org>

aparcar commented:
@tmn505 I rebased your code and it builds fine. Ideally I'd create a hybrid patch combining the best bits of @lynxis and @tmn505 .
Right now there is likely an issue with the UUID which I'll try to fix

    [    1.488489] Waiting for root device PARTUUID=fc80ef2a-02...

aparcar commented on target/linux/x86/image/64.mk:
```diff
@@ -0,0 +1,5 @@
+define Device/generic
```
```suggestion
define Device/generic-x86_64
```

This way the x86 target would have unique IDs for PROFILES. I'd be in favor of this. So you can just specify the desired PROFILE to build and the target is deterministic

aparcar commented:
Superseded via https://patchwork.ozlabs.org/project/openwrt/list/?series=165613
